### ISSUE TITLE
JUnit tests are broken

### ISSUE THREAD

**Issue by barbeau** [barbeau/gpstest#451]
**Describe the bug**
If you run the JUnit tests in Android Studio 4.1, you get this:

~~~
"C:\Program Files\Android\Android Studio\jre\bin\java.exe" -ea -Didea.test.cyclic.buffer.size=1048576 "-javaagent:C:\Program Files\Android\Android Studio\lib\idea_rt.jar=58159:C:\Program Files\Android\Android Studio\bin" -Dfile.encoding=UTF-8 -classpath "C:\Program Files\Android\Android Studio\lib\idea_rt.jar;C:\Program Files\Android\Android Studio\plugins\junit\lib\junit5-rt.jar;C:\Program Files\Android\Android Studio\plugins\junit\lib\junit-rt.jar;C:\Users\sjbar\AppData\Local\Android\sdk\platforms\android-30\data\res;C:\android-projects\gpstest\GPSTest\build\intermediates\javac\googleDebugUnitTest\classes;C:\android-projects\gpstest\GPSTest\build\intermediates\javac\googleDebug\classes;C:\android-projects\gpstest\GPSTest\build\tmp\kotlin-classes\googleDebugUnitTest;C:\android-projects\gpstest\GPSTest\build\intermediates\unit_test_config_directory\googleDebugUnitTest\out;C:\android-projects\gpstest\GPSTest\build\intermediates\compile_and_runtime_not_namespaced_r_class_jar\googleDebug\R.jar;C:\android-projects\gpstest\GPSTest\build\intermediates\compile_and_runtime_not_namespaced_r_class_jar\googleDebugAndroidTest\R.jar;C:\android-projects\gpstest\GPSTest\build\tmp\kotlin-classes\googleDebugAndroidTest;C:\android-projects\gpstest\GPSTest\build\tmp\kotlin-classes\googleDebug;C:\Users\sjbar\AppData\Local\Android\sdk\platforms\android-30\optional\android.test.mock.jar;C:\Users\sjbar\AppData\Local\Android\sdk\platforms\android-30\optional\android.test.base.jar;C:\Users\sjbar\AppData\Local\Android\sdk\platforms\android-30\optional\android.test.runner.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\junit\junit\4.13.1\cdd00374f1fee76b11e2a9d127405aa3f6be5b6a\junit-4.13.1.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\org.hamcrest\hamcrest-core\1.3\42a25dc3219429f0e5d060061f71acb49bf010a0\hamcrest-core-1.3.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\67f8eb4c60be3966260ffbefaa168cdc\jetified-core-1.3.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\67f8eb4c60be3966260ffbefaa168cdc\jetified-core-1.3.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\0a121adf2e994b5868eddf70fb054cb4\rules-1.3.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\0a121adf2e994b5868eddf70fb054cb4\rules-1.3.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\162604a31220820f43fa75a092fdf681\runner-1.3.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\162604a31220820f43fa75a092fdf681\runner-1.3.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\02ba3c2143cad709736e0f094b08e0bb\core-testing-2.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\02ba3c2143cad709736e0f094b08e0bb\core-testing-2.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f2a382809a95e340c5cb03e2ef4c9544\multidex-instrumentation-2.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f2a382809a95e340c5cb03e2ef4c9544\multidex-instrumentation-2.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\025c30996645da391f6b7dc43df11c62\monitor-1.3.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\025c30996645da391f6b7dc43df11c62\monitor-1.3.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\androidx.collection\collection\1.1.0\1f27220b47669781457de0d600849a5de0e89909\collection-1.1.0.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\androidx.arch.core\core-common\2.1.0\b3152fc64428c9354344bd89848ecddc09b6f07e\core-common-2.1.0.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\androidx.lifecycle\lifecycle-common\2.2.0\4ef09a745007778eef83b92f8f23987a8ea59496\lifecycle-common-2.2.0.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\androidx.annotation\annotation\1.1.0\e3a6fb2f40e3a3842e6b7472628ba4ce416ea4c8\annotation-1.1.0.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\androidx.constraintlayout\constraintlayout-solver\2.0.0-rc1\8e963a1952dd86ecccb463c0f8396246e51b2a92\constraintlayout-solver-2.0.0-rc1.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlin\kotlin-stdlib-jdk7\1.4.10\30e46450b0bb3dbf43898d2f461be4a942784780\kotlin-stdlib-jdk7-1.4.10.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlinx\kotlinx-coroutines-android\1.3.9\df17db5e329363c4e9cc7bf5b661ce3723a3e460\kotlinx-coroutines-android-1.3.9.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlinx\kotlinx-coroutines-core-jvm\1.3.9\4be434f5e86c1998a273e7f19a7286440894f0b0\kotlinx-coroutines-core-jvm-1.3.9.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlin\kotlin-stdlib\1.4.10\ea29e063d2bbe695be13e9d044dcfb0c7add398e\kotlin-stdlib-1.4.10.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\org.jetbrains.kotlin\kotlin-stdlib-common\1.4.10\6229be3465805c99db1142ad75e6c6ddeac0b04c\kotlin-stdlib-common-1.4.10.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\org.jetbrains\annotations\13.0\919f0dfe192fb4e063e7dacadee7f8bb9a2672a9\annotations-13.0.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\com.google.zxing\android-integration\3.3.0\e0f3e0def130c30c1fab6f891cf52fb01cc8522b\android-integration-3.3.0.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\commons-io\commons-io\2.6\815893df5f31da2ece4040fe0a12fd44b577afaf\commons-io-2.6.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\com.fasterxml.jackson.core\jackson-databind\2.11.2\ee08bbd8975dde844307fe8309dfcd5ec7ee129d\jackson-databind-2.11.2.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\com.fasterxml.jackson.core\jackson-core\2.11.2\bc022ab0f0c83c07f9c52c5ab9a6a4932b15cc35\jackson-core-2.11.2.jar;C:\Users\sjbar\.gradle\caches\modules-2\files-2.1\com.fasterxml.jackson.core\jackson-annotations\2.11.2\e0a7f61fce3e3eac38a079c11831868269de2ea\jackson-annotations-2.11.2.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\3d8221ae5217d0626a93bed64411d710\multidex-2.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\3d8221ae5217d0626a93bed64411d710\multidex-2.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f5d96d626385d523bf83de715d9212ff\jetified-play-services-maps-16.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f5d96d626385d523bf83de715d9212ff\jetified-play-services-maps-16.1.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f5d96d626385d523bf83de715d9212ff\jetified-play-services-maps-16.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\01fb93e1dbff66fa8460bf77acaeaa14\jetified-play-services-base-16.0.1\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\01fb93e1dbff66fa8460bf77acaeaa14\jetified-play-services-base-16.0.1\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\01fb93e1dbff66fa8460bf77acaeaa14\jetified-play-services-base-16.0.1\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\12acbe78c97e745b60e51e7ffdecc7f5\jetified-play-services-tasks-16.0.1\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\12acbe78c97e745b60e51e7ffdecc7f5\jetified-play-services-tasks-16.0.1\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\2a805ee5d7d09c132ee5a010ebf21800\jetified-play-services-basement-16.0.1\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\2a805ee5d7d09c132ee5a010ebf21800\jetified-play-services-basement-16.0.1\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\2a805ee5d7d09c132ee5a010ebf21800\jetified-play-services-basement-16.0.1\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\00922e5a2c4fcabb8eaf998b09e67e21\jetified-library-3.4.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\00922e5a2c4fcabb8eaf998b09e67e21\jetified-library-3.4.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\00922e5a2c4fcabb8eaf998b09e67e21\jetified-library-3.4.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\36c043c032eaac21f8f4a23f521b220d\legacy-support-v4-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\36c043c032eaac21f8f4a23f521b220d\legacy-support-v4-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f1039009e1f7a4d8d242c65532a68282\material-1.2.1\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f1039009e1f7a4d8d242c65532a68282\material-1.2.1\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f1039009e1f7a4d8d242c65532a68282\material-1.2.1\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\0e736fec42893c7f34d4f9101454df88\constraintlayout-2.0.0-rc1\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\0e736fec42893c7f34d4f9101454df88\constraintlayout-2.0.0-rc1\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\0e736fec42893c7f34d4f9101454df88\constraintlayout-2.0.0-rc1\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f65f3a6766d90195124af3fefb344479\appcompat-1.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f65f3a6766d90195124af3fefb344479\appcompat-1.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\f65f3a6766d90195124af3fefb344479\appcompat-1.1.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\bde568db7f19a9667eb777534776246e\jetified-viewpager2-1.0.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\bde568db7f19a9667eb777534776246e\jetified-viewpager2-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\bde568db7f19a9667eb777534776246e\jetified-viewpager2-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\5c54c45f5ac4048a39afb1cdf2268935\recyclerview-1.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\5c54c45f5ac4048a39afb1cdf2268935\recyclerview-1.1.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\5c54c45f5ac4048a39afb1cdf2268935\recyclerview-1.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\6b4b738ac9984e5ec2af237a5afcf2bf\jetified-core-ktx-1.3.2\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\6b4b738ac9984e5ec2af237a5afcf2bf\jetified-core-ktx-1.3.2\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\1954bb79850b1f9b44a19a8f8f1079ab\lifecycle-extensions-2.1.0-beta01\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\1954bb79850b1f9b44a19a8f8f1079ab\lifecycle-extensions-2.1.0-beta01\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e320b4bd49f3909943a79e25b13ac56d\fragment-1.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e320b4bd49f3909943a79e25b13ac56d\fragment-1.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\0153ef955b9796bd1e337ccef391362b\jetified-appcompat-resources-1.1.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\0153ef955b9796bd1e337ccef391362b\jetified-appcompat-resources-1.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\0153ef955b9796bd1e337ccef391362b\jetified-appcompat-resources-1.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e47ad70f535ad29c20cca4f08c350909\legacy-support-core-ui-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e47ad70f535ad29c20cca4f08c350909\legacy-support-core-ui-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\cca21dddeadf8a9223d20932941e89d5\drawerlayout-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\cca21dddeadf8a9223d20932941e89d5\drawerlayout-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\925766f15736023e891d7776ae4309e6\coordinatorlayout-1.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\925766f15736023e891d7776ae4309e6\coordinatorlayout-1.1.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\925766f15736023e891d7776ae4309e6\coordinatorlayout-1.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\63f4b5d699361818de2d669e8bbbdee5\transition-1.2.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\63f4b5d699361818de2d669e8bbbdee5\transition-1.2.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\63f4b5d699361818de2d669e8bbbdee5\transition-1.2.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\725041083447a9143eb8cb9d5c229aa4\vectordrawable-animated-1.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\725041083447a9143eb8cb9d5c229aa4\vectordrawable-animated-1.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\aa4fabacd2c30af5b5bdef1f5a36c0de\vectordrawable-1.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\aa4fabacd2c30af5b5bdef1f5a36c0de\vectordrawable-1.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ac40ff9acdb830edef055a93aa6af444\viewpager-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ac40ff9acdb830edef055a93aa6af444\viewpager-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\040099834b582bec7a5958cfcf954cda\slidingpanelayout-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\040099834b582bec7a5958cfcf954cda\slidingpanelayout-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\9a66ec3bb14729873bebeb1056f18fe2\customview-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\9a66ec3bb14729873bebeb1056f18fe2\customview-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\217f3373a7d3f2c22d16417480a006d3\legacy-support-core-utils-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\217f3373a7d3f2c22d16417480a006d3\legacy-support-core-utils-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\7290f53958304a0c71bbe043bd01330e\loader-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\7290f53958304a0c71bbe043bd01330e\loader-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\86cd42cca04559997b9e79660d1bc7a5\jetified-activity-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\86cd42cca04559997b9e79660d1bc7a5\jetified-activity-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\92f5c8798b500abf18effb458f32e58b\media-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\92f5c8798b500abf18effb458f32e58b\media-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\92f5c8798b500abf18effb458f32e58b\media-1.0.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\2590614af03719ba9b66784129040772\swiperefreshlayout-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\2590614af03719ba9b66784129040772\swiperefreshlayout-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\d667dc41f87e62bbe5b41566b12f5a99\asynclayoutinflater-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\d667dc41f87e62bbe5b41566b12f5a99\asynclayoutinflater-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ad964c660487516315ee8e3a5d93a8c3\core-1.3.2\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ad964c660487516315ee8e3a5d93a8c3\core-1.3.2\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ad964c660487516315ee8e3a5d93a8c3\core-1.3.2\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\69a7c80e9a56b90f03ab2c7e5fd8556b\cardview-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\69a7c80e9a56b90f03ab2c7e5fd8556b\cardview-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\69a7c80e9a56b90f03ab2c7e5fd8556b\cardview-1.0.0\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ef308f223e635330c77b74537fee4557\jetified-lifecycle-runtime-ktx-2.2.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ef308f223e635330c77b74537fee4557\jetified-lifecycle-runtime-ktx-2.2.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\fe5a9c778f8a4fdd24fcded7d6a7e589\cursoradapter-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\fe5a9c778f8a4fdd24fcded7d6a7e589\cursoradapter-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\5fc5d8e3d6e254faa6d706084c638623\versionedparcelable-1.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\5fc5d8e3d6e254faa6d706084c638623\versionedparcelable-1.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\6dca8ca66ab0704f15137824d1856ae5\jetified-lifecycle-process-2.1.0-beta01\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\6dca8ca66ab0704f15137824d1856ae5\jetified-lifecycle-process-2.1.0-beta01\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ab262baea7e1f0728caacaf67628ff5c\jetified-lifecycle-service-2.1.0-beta01\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\ab262baea7e1f0728caacaf67628ff5c\jetified-lifecycle-service-2.1.0-beta01\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\65a9b13ec0ec0ed656b43462d5d7dc6e\lifecycle-runtime-2.2.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\65a9b13ec0ec0ed656b43462d5d7dc6e\lifecycle-runtime-2.2.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\a0d63fc0880e9c828b7dd01557cf6178\lifecycle-livedata-2.1.0-beta01\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\a0d63fc0880e9c828b7dd01557cf6178\lifecycle-livedata-2.1.0-beta01\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\97c782c299f2026ef7e49c846ef0894f\lifecycle-livedata-core-2.1.0-beta01\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\97c782c299f2026ef7e49c846ef0894f\lifecycle-livedata-core-2.1.0-beta01\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\650cd90c1158ffc8b997f4e7a1d280a6\core-runtime-2.1.0-beta01\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\650cd90c1158ffc8b997f4e7a1d280a6\core-runtime-2.1.0-beta01\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\be7f8916c85242be4375c60b7f795a49\jetified-savedstate-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\be7f8916c85242be4375c60b7f795a49\jetified-savedstate-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\88698de896f26102d261914a393c5dde\lifecycle-viewmodel-2.1.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\88698de896f26102d261914a393c5dde\lifecycle-viewmodel-2.1.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e3945bcef529ba35cd7923313474a4b7\interpolator-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e3945bcef529ba35cd7923313474a4b7\interpolator-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\a3267bb05e8ea650c6b99e7054235377\documentfile-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\a3267bb05e8ea650c6b99e7054235377\documentfile-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\fe78902ba549d15f4f8cf9083fbb66ec\localbroadcastmanager-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\fe78902ba549d15f4f8cf9083fbb66ec\localbroadcastmanager-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\d9e0e7e1757141c4c6284d753f45dfe4\print-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\d9e0e7e1757141c4c6284d753f45dfe4\print-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e556565688b6e8d8d7487a8455ccade2\jetified-android-maps-utils-0.5\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e556565688b6e8d8d7487a8455ccade2\jetified-android-maps-utils-0.5\res;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\e556565688b6e8d8d7487a8455ccade2\jetified-android-maps-utils-0.5\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\da978c696891c94cfc03109378a36124\jetified-annotation-experimental-1.0.0\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\da978c696891c94cfc03109378a36124\jetified-annotation-experimental-1.0.0\AndroidManifest.xml;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\c2de503e7c323b1b2b39c3b1278f5565\jetified-MPAndroidChart-v3.1.0-alpha\jars\classes.jar;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\c2de503e7c323b1b2b39c3b1278f5565\jetified-MPAndroidChart-v3.1.0-alpha\AndroidManifest.xml;C:\android-projects\gpstest\GPSTest\build\intermediates\java_res\googleDebugUnitTest\out;C:\android-projects\gpstest\GPSTest\build\intermediates\java_res\googleDebug\out;C:\Users\sjbar\.gradle\caches\transforms-2\files-2.1\34105728e9b690f52106714b80226ab7\android.jar" com.intellij.rt.junit.JUnitStarter -ideVersion5 @w@C:\Users\sjbar\AppData\Local\Temp\idea_working_dirs_junit.tmp @C:\Users\sjbar\AppData\Local\Temp\idea_junit.tmp -socket58158
!!! JUnit version 3.8 or later expected:

java.lang.RuntimeException: Stub!
	at junit.runner.BaseTestRunner.<init>(BaseTestRunner.java:7)
	at junit.textui.TestRunner.<init>(TestRunner.java:54)
	at junit.textui.TestRunner.<init>(TestRunner.java:48)
	at junit.textui.TestRunner.<init>(TestRunner.java:41)
	at com.intellij.rt.junit.JUnitStarter.junitVersionChecks(JUnitStarter.java:212)
	at com.intellij.rt.junit.JUnitStarter.canWorkWithJUnitVersion(JUnitStarter.java:195)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:49)
~~~

~~This seems to be the root of the same issue seen in this CI build, which is what happens when running `./gradlew test` from the command line:~~
https://travis-ci.org/github/barbeau/gpstest/jobs/745713794

~~~
> Task :GPSTest:testGoogleDebugUnitTest
com.android.gpstest.util.UIUtilsTest > testSnrToTextViewLeftMargin FAILED
    java.lang.NoClassDefFoundError at UIUtilsTest.java:85
        Caused by: java.lang.ClassNotFoundException at UIUtilsTest.java:85
~~~

~~It's strange because this suddenly broke with a commit in this PR  on CI and has been failing since, even though nothing should have changed that would cause this:~~
https://github.com/barbeau/gpstest/pull/448

(**EDIT** - CI failure appears unrelated - see below)

This seems to be the issue:
https://issuetracker.google.com/issues/170328018

...although I tested using Android Studio 4.2 Canary 16 and still had the same problem.

**To Reproduce**
1. Run JUnit tests from within Android Studio

**Expected behavior**
JUnit tests should execute and pass

**Observed behavior**
JUnit tests won't run

**App, Device and Android version:** 

N/A

**Comment by barbeau** [comment]
Running  `./gradlew testcheck connectedCheck` (after ignoring lint checks see https://github.com/barbeau/gpstest/issues/452) works for me on my machine, so I'm not sure what's going on on Travis. Maybe a separate issue?

**Comment by barbeau** [comment]
Travis issue appears unrelated - see https://github.com/barbeau/gpstest/pull/448 for notes on Travis issue.

**Comment by barbeau** [comment]
This was apparently fixed in AS 4.1.2:
https://androidstudio.googleblog.com/2021/01/android-studio-412-available.html?m=1

I can confirm it works now with 4.1.2:

~~~
Android Studio 4.1.2
Build #AI-201.8743.12.41.7042882, built on December 19, 2020
Runtime version: 1.8.0_242-release-1644-b01 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1546M
Cores: 8
Registry: ide.new.welcome.screen.force=true, external.system.auto.import.disabled=true
Non-Bundled Plugins: org.intellij.plugins.markdown, com.intellij.marketplace, org.jetbrains.kotlin
~~~
